samples/client/petstore/ruby-faraday/lib/petstore/models/map_test.rb

```diff
@@ -119,13 +119,15 @@ def initialize(attributes = {})
     # Show invalid properties with the reasons. Usually used together with valid?
     # @return Array for valid properties with the reasons
     def list_invalid_properties
+      warn '[DEPRECATED] the `list_invalid_properties` method is obsolete'
       invalid_properties = Array.new
       invalid_properties
     end
 
     # Check to see if the all the properties in the model are valid
     # @return true if the model is valid
     def valid?
+      warn '[DEPRECATED] the `valid?` method is obsolete'
```

Comment:
the oneOf, anyOf implementation is still using this method so we need to update the implementation accordingly to replace `valid?` with something else.

Reply:
I believe the reference to `valid?` on line [126](https://github.com/OpenAPITools/openapi-generator/blob/bfcd6463563598d99545ae1185e60a43969a6f59/modules/openapi-generator/src/main/resources/ruby-client/partial_oneof_module.mustache#L126) can be removed. Since the `model` is created from the `build_from_hash` method, which now fully supports attribute validation, any model created from that method should be guaranteed to be valid. Let me know if this line can be updated to `return model if model`

Reply:
>  in which case, I'll add enum validation to the attribute writers.

Yes, I think we need enum validation as well in attribute writers.

I've filed https://github.com/OpenAPITools/openapi-generator/pull/16137 to add some tests for oneOf. Will merge your change into my PR to see if there's any change in oneOf model behaviour.

Reply:
I tried removing the `valid?` check in oneOf template and the tests also passed so I think we are in good shape to mark `valid?` as deprecated

Reply:
FYI. https://github.com/OpenAPITools/openapi-generator/pull/16137 has been merged.

We can remove `valid?` later in oneOf template

Reply:
> Yes, I think we need enum validation as well in attribute writers.

Please submit another PR for this as I'll merge this for the time being since I've some other enhancements for Ruby models as well.

Thanks again for the PR.

Reply:
UPDATE: filed https://github.com/OpenAPITools/openapi-generator/pull/16147 to remove `invalid?` in oneOf template and add anyOf support.